src/net_processing.cpp

```diff
@@ -1330,7 +1330,7 @@ void PeerLogicValidation::UpdatedBlockTip(const CBlockIndex *pindexNew, const CB
         connman->ForEachNode([nNewHeight, &vHashes](CNode* pnode) {
             if (nNewHeight > (pnode->nStartingHeight != -1 ? pnode->nStartingHeight - 2000 : 0)) {
                 for (const uint256& hash : reverse_iterate(vHashes)) {
-                    pnode->PushBlockHash(hash);
+                    WITH_LOCK(pnode->cs_inventory, pnode->vBlockHashesToAnnounce.push_back(hash));
```

Comment:
When doing this anyway, woulid it make sense to move this lock outside the loop? Re-taking it for every `node × every block` seems overkill.

Reply:
Good idea. Done.

Reply:
e335e4fcc881260906fb561f23a1b5017ed04d91

The suggestion was done in 6db15a8b79597a0f311490acd89ee50db298c6e8, but IMO should be in e335e4fcc881260906fb561f23a1b5017ed04d91.

Reply:
Done.